### DISCUSSION TITLE
Custom Lexicons: Publishing records with a custom schema to a server you don't own

### DISCUSSION THREAD

**Discussion by ebwinters** [bluesky-social/atproto#3116] [Clarifications and "Why?"] [answered]
Hi Everyone
I have gone through the [statusphere example](https://github.com/bluesky-social/statusphere-example-app?tab=readme-ov-file) and a lot of the basic concepts make sense to me. One thing I cannot quite understand is how publishing events with the `xyz.statusphere.status` schema from localhost can push those events to Bluesky's servers, and ultimately a relay. 

1. Since Bluesky's servers have no concept of this status lexicon, how do they parse the event and store it in my repository?
2. Wouldn't allowing publishing behavior allow users to "flood" bluesky's PDS servers with junk events? What if I had a script to publish a record with a new lexicon every 1s? Wouldn't that lead to an explosion in collections in my repository hosted on Bluesky PDS?

**Comment by DavidBuchanan314** [comment, marked as answer]
1. You don't need the lexicon to parse a record, only to validate the schema. validation can be disabled.

2. Rate limits and account bans exist

**Reply by npetrangelo** [reply]
I'm still unclear on this, how does a PDS know which data to respond with when it gets a request with a custom lexicon? Actually moreover, if I want a specific piece of data, how should I request it when the PDS is lexicon agnostic?

**Reply by DavidBuchanan314** [reply]
I don't really understand the question, the PDS is indeed lexicon agnostic so there's no special logic required, it responds to all requests the same regardless of lexicon (aside from a few special cases)

**Reply by bnewbold** [reply]
The PDS hosts atproto repos. atproto repos are similar to git repos, or just a .zip archive. You can put any kind of file in git, and you can put basically any kind of JSON data in an atproto repo. There are rate-limits of how many files you can create in a given time frame, but no limits on what types of files.

Bluesky posts are just a one kind of file. [whtwnd.com w](https://whtwnd.com/) blog posts are a different kind of file.
